modules/internal/ChapelArray.chpl

```diff
@@ -166,6 +166,7 @@ module ChapelArray {
   use ArrayViewSlice;
   use ArrayViewRankChange;
   use ArrayViewReindex;
+  use ExternalArray;
```

Comment:
I was mostly keeping it close to the other _getArray function since they had a similar origin, but I can make it local to the ExternalArray module (and insert uses when compiling with --library mode)

Reply:
I'm mostly intending this type to be under the covers, so a user could say:
```chapel
export proc foo(x: [0..3] int) { ... }
```
and we would insert the transformation from what was actually provided to us into a form that the original foo would understand.  If ExternalArray is not included by default, then we would also need to insert a `use` in the module for this transformation to work, right?

Reply:
That's cool, but let's leave it as a non-default-included module until such time as the compiler is using it.

Reply:
If I leave it in modules/internal, a use statement has difficulty finding it (I think because we assume everything in modules/internal is `use`d by ChapelStandard or something else), so I'm moving it to modules/packages for now.